packages/plugins/accounts-manager/src/index.ts

```diff
@@ -73,87 +34,28 @@ export default class AccountsManager {
-    }
-    if (params.reqData.method === blockchainConstants.transactionMethods.eth_sendTransaction && this.accounts.length) {
-      // Check if we have that account in our wallet
-      const account = this.accounts.find((acc) => Web3.utils.toChecksumAddress(acc.address) === Web3.utils.toChecksumAddress(params.reqData.params[0].from));
-      if (account && account.privateKey) {
-        return this.signTransactionQueue.push({ payload: params.reqData.params[0], account }, (err: any, newPayload: any) => {
-          if (err) {
-            return callback(err, null);
-          }
-          params.reqData.method = blockchainConstants.transactionMethods.eth_sendRawTransaction;
-          params.reqData.params = [newPayload];
-          callback(err, params);
-        });
-      }
-    }
-    callback(null, params);
-  }
-
-  private async checkBlockchainResponse(params: any, callback: (error: any, result: any) => void) {
-    if (!this.ready) {
-      return callback(null, params);
-    }
-    if ((params.reqData.method === blockchainConstants.transactionMethods.eth_accounts ||
-      params.reqData.method === blockchainConstants.transactionMethods.personal_listAccounts) && this.accounts.length) {
-      if (!arrayEqual(params.respData.result, this.nodeAccounts)) {
-        this.nodeAccounts = params.respData.result;
+  get accounts() {
+    return (async () => {
+      if (!this._accounts) {
         const web3 = await this.web3;
-        this.accounts = AccountParser.parseAccountsConfig(this.embark.config.blockchainConfig.accounts, web3, dappPath(), this.logger, this.nodeAccounts);
+        this._accounts = await web3.eth.getAccounts();
       }
-      params.respData.result = this.accounts.map((acc) => {
-        if (acc.address) {
-          return acc.address;
-        }
-        return acc;
-      });
-      return callback(null, params);
-    }
-    callback(null, params);
+      return this._accounts || [];
+    })();
   }
 
-  private async parseAndFundAccounts(accounts: any[] | null) {
+  private async parseAndFundAccounts() {
     const web3 = await this.web3;
+    const accounts = await this.accounts;
 
-    const nodeAccounts = await web3.eth.getAccounts();
-    this.nodeAccounts = nodeAccounts;
-    this.accounts = AccountParser.parseAccountsConfig(accounts || this.embark.config.blockchainConfig.accounts, web3, dappPath(), this.logger, nodeAccounts);
-
-    if (!this.accounts.length || !this.embark.config.blockchainConfig.isDev) {
-      this.ready = true;
+    if (!accounts.length || !this.embark.config.blockchainConfig.isDev) {
       return;
     }
     try {
       const coinbase = await web3.eth.getCoinbase();
-      const accts = this.accounts
-        .filter((account) => account.address);
 
-      async.eachLimit(accts, 1, (acct, eachCb) => {
-        fundAccount(web3, acct.address, coinbase, acct.hexBalance)
+      async.eachLimit(accounts, 1, (acct, eachCb) => {
+        fundAccount(web3, acct, coinbase)
```

Comment:
As @PascalPrecht said, we used to rely on accounts returned from `parseAccountsConfig`, but now we only get the addresses. That's a problem, because we no longer fund the accounts up to the balance that they require. This will cause some tests to fail.

I don't remember the reason why the parsing was moved to another module, but if this module is the accounts manager, I would expect it to be the source of the accounts, now it's more like the "account funder", and even at that, it doesn't perform correctly.

Maybe we move back the parsing here and have the other modules get the accounts from here (but it's a plugin, so probably doesn't work), or we also put back the RPC call changes.
We can also have this module get the accounts from the other, but I think the other is a plugin too.

Reply:
So with the `rpc-manager` in place and the proxy working correctly, a call to `web3.eth.getAccounts()` should get the correct accounts. No need to have a module being the source of truth.